platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/tipset/TipsetScoreCalculator.java

```diff
@@ -0,0 +1,281 @@
+
+    /**
+     * The number of snapshots to keep in {@link #snapshotHistory}.
+     */
+    private final int maxSnapshotHistorySize;
+
+    /**
+     * The total weight of all nodes.
+     */
+    private final long totalWeight;
+
+    /**
+     * The weight of the node tracked by this window.
+     */
+    private final long selfWeight;
+
+    /**
+     * The maximum possible advancement weight for an event.
+     */
+    private final long maximumPossibleAdvancementWeight;
+
+    /**
+     * The previous tipset advancement weight.
+     */
+    private TipsetAdvancementWeight previousScore = ZERO_ADVANCEMENT_WEIGHT;
+
+    /**
+     * The tipset of the latest self event, or the starting snapshot if there has not yet been a self event.
+     */
+    private Tipset latestSelfEventTipset;
+
+    /**
+     * Create a new tipset window.
+     *
+     * @param platformContext       the platform context
+     * @param addressBook           the current address book
+     * @param selfId                the ID of the node tracked by this window
+     * @param tipsetTracker         builds tipsets for individual events
+     * @param childlessEventTracker tracks non-ancient events without children
+     */
+    public TipsetScoreCalculator(
+            @NonNull final PlatformContext platformContext,
+            @NonNull final AddressBook addressBook,
+            @NonNull final NodeId selfId,
+            @NonNull final TipsetTracker tipsetTracker,
+            @NonNull final ChildlessEventTracker childlessEventTracker) {
+
+        this.selfId = Objects.requireNonNull(selfId);
+        this.tipsetTracker = Objects.requireNonNull(tipsetTracker);
+        this.childlessEventTracker = Objects.requireNonNull(childlessEventTracker);
+        Objects.requireNonNull(addressBook);
+
+        totalWeight = addressBook.getTotalWeight();
+        selfWeight = addressBook.getAddress(selfId).getWeight();
+        maximumPossibleAdvancementWeight = totalWeight - selfWeight;
+        maxSnapshotHistorySize = platformContext
+                .getConfiguration()
+                .getConfigData(EventCreationConfig.class)
+                .tipsetSnapshotHistorySize();
+
+        snapshot = new Tipset(addressBook);
+        latestSelfEventTipset = snapshot;
+        snapshotHistory.add(snapshot);
+    }
+
+    /**
+     * Get the maximum possible tipset score that a new event can achieve.
+     */
+    public long getMaximumPossibleScore() {
+        return maximumPossibleAdvancementWeight;
+    }
+
+    /**
+     * Get the current tipset snapshot.
+     *
+     * @return the current tipset snapshot
+     */
+    public @NonNull Tipset getSnapshot() {
+        return snapshot;
+    }
+
+    /**
+     * Add an event created by this node and compute the increase in tipset score. Higher score changes mean that this
+     * event caused consensus to advance more. A score change of 0 means that this event did not advance consensus. A
```

Comment:
Use future tense here instead, because it _will_ allow consensus to advance more, but does not actually calculate consensus.

Reply:
apply throughout the javadoc

Reply:
done